otwcode/otwarchive
# AO3-5721 Delete challenge without JavaScript

## Issue

https://otwarchive.atlassian.net/browse/AO3-5721

## Purpose

A challenge mod can delete a challenge either on the Challenge Settings page, or on the profile of the collection that the challenge belongs to. Both buttons ("Delete Challenge") trigger a JavaScript function that shows a warning prompt and, after user confirmation, deletes the challenge. Users who have JavaScript disabled, however, receive a 404 error. They can't delete their challenge.

The suggested change adds a fallback option for users who have JavaScript disabled. These users get routed to a page where they can confirm that they want to delete the challenge by submitting a form. Users with JavaScript enabled can still delete via JavaScript prompt.

This pull request further contains added steps in the Gift Exchange and the Prompt Meme Features to test this new/ changed functionality.

I forgot to add the ticket number to two of my commit messages. :( Sorry!

## Testing Instructions

1. Turn off JavaScript in your browser
2. Create a collection with a Prompt Meme challenge
3. Go to your new collection's profile page and press "Delete Challenge"

Expectation: You should be routed to a page whose URL ends with /confirm_delete where you're asked if you are sure that you want to delete your challenge.

4. Press "Yes, Delete Challenge".

Expectation: You should now be routed to your collection page with a notice at the top saying "Challenge settings were deleted".

5. In your collection, create another Prompt Meme challenge
6. After saving your challenge settings, return to your Challenge Settings page
7. In the top right corner, press "Delete Challenge"

Expectation: see above after step 3

8. Continue with step 4 until you see "Challenge settings were deleted."

The same should be tested for Gift Exchange challenges.

Finally, challenge deletion with JavaScript enabled should be re-tested. Do the following for both challenge types, Prompt Memes and Gift Exchanges. For each type of challenge, test both the button on the collection profile page and the button on the Challenge Settings page:

1. Make sure JavaScript is enabled in your browser
2. Create a collection and a challenge, Prompt Meme or Gift Exchange
3. On the collection's profile page or on the Challenge Settings page, press "Delete Challenge"

Expectation: You should receive a prompt/ pop-up asking you if you want to delete your challenge.

4. Press "Okay"

Expectation: You should be routed to the collection page and see a notice "Challenge settings were deleted"

## Credit

teyla

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*

she or they
